addons/website_blog/controllers/main.py

```diff
@@ -83,19 +83,21 @@ def _prepare_blog_values(self, blogs=False, date_begin=False, date_end=False, ta
         else:
             domain += [("post_date", "<=", fields.Datetime.now())]
 
+        show_teaser = request.website.viewref('website_blog.opt_blog_show_teaser').active
+        fullwidth_cover = request.website.viewref('website_blog.opt_blog_cover_post_fullwidth_design').active
         total = BlogPost.search_count(domain)
-        posts = BlogPost.search(domain, offset=(page - 1) * self._blog_post_per_page, limit=self._blog_post_per_page, order="post_date, id asc")
+        if not fullwidth_cover and show_teaser:
+            posts = BlogPost.search(domain, offset=(page - 1) * self._blog_post_per_page + 1, limit=self._blog_post_per_page, order="post_date desc, id asc")
+        else:
+            posts = BlogPost.search(domain, offset=(page - 1) * self._blog_post_per_page, limit=self._blog_post_per_page, order="post_date desc, id asc")
         pager = request.website.pager(
             url=request.httprequest.path.partition('/page/')[0],
             total=total,
             page=page,
             step=self._blog_post_per_page,
         )
 
-        first_post = next((p for p in posts if p.website_published), None)
-        first_cover = first_post and json.loads(first_post.cover_properties)
-        covers = [json.loads(b.cover_properties) for b in posts]
-
+        first_post = next((p for p in BlogPost.search(domain, limit=1, order="post_date desc, id asc") if p.website_published), None)
```

Comment:
hmmmm

`next((p for p in BlogPost.search(domain, limit=1, order="post_date desc, id asc") if p.website_published), None)`

will not work if the first is no website_published... you will have None instead of the second

btw, `next(search(limit=1))` of 1 record id useless

btw make `a search(condA) if condB` => `search(condA et condB)`

btw just keep old code, but use `blogs[1:] `or `blogs` depending of show_teaser or not

:/